### DISCUSSION TITLE
Connect to MySql database

### DISCUSSION THREAD

**Discussion by GeertWil** [sshnet/SSH.NET#1435] [Q&A] [answered]
My hosting provider requires a SSH tunnel to connect to my MySql database. I want to access the database using a program written in C#.  The program is for .NET Framework 4.8.1. I'm used to connect to MySql databases in C#, but never via SSH.
I have set up and installed key pair. It looks like I can succesfully make a SSH connection, but the connection to the MySql database doesn't seem to work (Error: Can't make a connection to any of the specified MySQL hosts). Can someone point me to some C# code to successfuly make a MySql connection using SSH.NET?

**Comment by lifeincha0s** [comment, marked as answer]
You could try port forwarding. You can set up the initial SSH connection to the database server and use it as the encrypted tunnel. Afterwards you can just add a forwarded port to the remote server and connect locally to the database port. I only recently started using local port forwarding for my app to pass through a bastion host. I am no pro and wouldnt know which type of forwarding would fit your situation, maybe dynamic or remote. It is something you could look into, just a thought.

**Reply by GeertWil** [reply]
It did cost me pints of blood, sweat and tears, but I finally got it working. Port forwarding was indeed the solution, but I was trying that already. The problem was a port confict with my local MySql server.  Thanks for the help.

**Reply by lifeincha0s** [reply]
Well, that is good to hear. 👍
